spring-cloud-starter-task-composedtaskrunner/src/test/java/org/springframework/cloud/task/app/composedtaskrunner/support/OnOAuth2ClientCredentialsEnabledTests.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.task.app.composedtaskrunner.support;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+import org.springframework.boot.test.util.TestPropertyValues;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * @author Gunnar Hillert
+ */
+public class OnOAuth2ClientCredentialsEnabledTests {
+
+	@Test
+	public void noPropertySet() throws Exception {
+		AnnotationConfigApplicationContext context = load(Config.class);
+		assertThat(context.containsBean("myBean"), equalTo(false));
+		context.close();
+	}
+
+	@Test(expected=IllegalStateException.class)
+	public void propertySecurityOauth() throws Exception {
+		load(Config.class, "oauth2-client-credentials");
+	}
+
+	@Test
+	public void propertyClientId() throws Exception {
+		AnnotationConfigApplicationContext context = load(Config.class, "oauth2-client-credentials.client-id:12345");
+		assertThat(context.containsBean("myBean"), equalTo(true));
+		context.close();
+	}
+
+	@Test
+	public void clientIdOnlyWithNoValue() throws Exception {
+		AnnotationConfigApplicationContext context = load(Config.class, "oauth2-client-credentials.client-id");
+		assertThat(context.containsBean("myBean"), equalTo(true));
+		context.close();
```

Comment:
May want to have context.close in a `@After` so that if the assert throws exception the context will close.

Reply:
Done.